pkg/scheduler/admission/podnodeconstraints/admission.go

```diff
@@ -0,0 +1,84 @@
+package podnodeconstraints
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"reflect"
+
+	configlatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	"github.com/openshift/origin/pkg/scheduler/admission/podnodeconstraints/api"
+
+	admission "k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+func init() {
+	admission.RegisterPlugin("PodNodeConstraints", func(c kclient.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig, err := readConfig(config)
+		if err != nil {
+			return nil, err
+		}
+		return newPodNodeConstraints(pluginConfig), nil
+	})
+}
+
+func newPodNodeConstraints(config *api.Config) admission.Interface {
+	return &podNodeConstraints{
+		config:  config,
+		Handler: admission.NewHandler(admission.Create),
+	}
+}
+
+type podNodeConstraints struct {
+	*admission.Handler
+	config *api.Config
+}
+
+func readConfig(reader io.Reader) (*api.Config, error) {
+	if reader == nil || reflect.ValueOf(reader).IsNil() {
+		return &api.Config{}, nil
+	}
+
+	configBytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+	config := &api.Config{}
+	err = configlatest.ReadYAML(configBytes, config)
+	if err != nil {
+		return nil, err
+	}
+	// No validation needed since config is just 2 bools
+	return config, nil
+}
+
+func (oo *podNodeConstraints) Admit(aa admission.Attributes) error {
+	if oo.config == nil {
+		return nil
+	}
+	if aa.GetResource() != kapi.Resource("pods") {
```

Comment:
@sosiouxme made me change these separate checks to a single statement... I guess you should do the same 
https://github.com/openshift/origin/blob/master/pkg/quota/admission/runonceduration/admission.go#L69-L75